### PR TITLE
[Primitives v8] Add primitives section to Foundations

### PR DESCRIPTION
**Pull request by langermank**
Part one of color docs ✨ 

Adds a `Primitives` section to Foundations, with the following new pages:

- Getting started
- Token names
- Color
- Typography
- Size

Color, typography and size are all referencing Storybook tables for easy copy of CSS vars. Token names extends a few existing ADRs into public documentation.

### CODE REVIEW

**Review by joshblack — APPROVED** [review]
Loved the token names explainer, felt very clear to me. The images really helped with understanding how the different groups apply to the token name. I also really appreciated the sections walking through available values for each group 🔥 

Separately, noticed this super weird nav bug 🤔 No idea if it's related to this PR but wanted to share just in case
The navigation for a color section seems to be side by side instead of each section appearing one after the other vertically

File: content/foundations/primitives/color.mdx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+---
+
+<Note>The following color tokens are still in beta and require a fallback value be used.</Note>
+
+## Foundational
+
+### Foreground
+
+<StorybookEmbed
+  baseUrl="https://primer-b5d9560b27-26615723.drafts.github.io/storybook"
```
Comment (langermank) [inline content/foundations/primitives/color.mdx:14]:
```suggestion
  baseUrl="https://primer.style/primitives/storybook/"
```

File: content/foundations/primitives/color.mdx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  stories={[{id: 'color-functional-tables--foreground'}]}
+  height="640"
+/>
+
+### Background
+
+<StorybookEmbed
+  baseUrl="https://primer-b5d9560b27-26615723.drafts.github.io/storybook"
```
Comment (langermank) [inline content/foundations/primitives/color.mdx:22]:
```suggestion
  baseUrl="https://primer.style/primitives/storybook/"
```

File: content/foundations/primitives/color.mdx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  stories={[{id: 'color-functional-tables--background'}]}
+  height="330"
+/>
+
+### Border
+
+<StorybookEmbed
+  baseUrl="https://primer-b5d9560b27-26615723.drafts.github.io/storybook"
```
Comment (langermank) [inline content/foundations/primitives/color.mdx:30]:
```suggestion
  baseUrl="https://primer.style/primitives/storybook/"
```

File: content/foundations/primitives/color.mdx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  stories={[{id: 'color-functional-tables--border'}]}
+  height="640"
+/>
+
+### Shadow
+
+<StorybookEmbed
+  baseUrl="https://primer-b5d9560b27-26615723.drafts.github.io/storybook"
```
Comment (langermank) [inline content/foundations/primitives/color.mdx:38]:
```suggestion
  baseUrl="https://primer.style/primitives/storybook/"
```

File: content/foundations/primitives/color.mdx [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+  stories={[{id: 'color-functional-tables--shadow'}]}
+  height="220"
+/>
+
+## Pattern
+
+### Button
+
+<StorybookEmbed
+  baseUrl="https://primer-b5d9560b27-26615723.drafts.github.io/storybook"
```
Comment (langermank) [inline content/foundations/primitives/color.mdx:48]:
```suggestion
  baseUrl="https://primer.style/primitives/storybook/"
```

File: content/foundations/primitives/color.mdx [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+  stories={[{id: 'color-patterns--button'}]}
+  height="640"
+/>
+
+### Control
+
+<StorybookEmbed
+  baseUrl="https://primer-b5d9560b27-26615723.drafts.github.io/storybook"
```
Comment (langermank) [inline content/foundations/primitives/color.mdx:56]:
```suggestion
  baseUrl="https://primer.style/primitives/storybook/"
```

File: content/foundations/primitives/color.mdx [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+  stories={[{id: 'color-patterns--control'}]}
+  height="640"
+/>
+
+### Focus
+
+<StorybookEmbed
+  baseUrl="https://primer-b5d9560b27-26615723.drafts.github.io/storybook"
```
Comment (langermank) [inline content/foundations/primitives/color.mdx:64]:
```suggestion
  baseUrl="https://primer.style/primitives/storybook/"
```

File: content/foundations/primitives/color.mdx [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+  stories={[{id: 'color-patterns--focus'}]}
+  height="220"
+/>
+
+### Overlay
+
+<StorybookEmbed
+  baseUrl="https://primer-b5d9560b27-26615723.drafts.github.io/storybook"
```
Comment (langermank) [inline content/foundations/primitives/color.mdx:72]:
```suggestion
  baseUrl="https://primer.style/primitives/storybook/"
```

File: content/foundations/primitives/token-names.mdx [hunk 9]
```diff
@@ -0,0 +1,174 @@
+---
+title: Token names
+description:
+  An explanation of Primer's design token naming convention
+---
+
+## Intro
+
+Naming things is hard. Primer design tokens are named using a consistent convention to make it easier to understand what they do and how they should be used at a glance.
```
Comment (langermank) [inline content/foundations/primitives/token-names.mdx:9]:
```suggestion
Primer design tokens are named using a consistent convention to make it easier to understand what they do and how they should be used at a glance.
```

**Comment by lukasoppermann** [comment]
I am assuming this error in the menu is not due to your changes, right?

CleanShot 2024-02-12 at 14 14 00@2x

**Review by lukasoppermann — APPROVED** [review]
💪

**Pull request merged**